caddy/app.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2024 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package caddy provides an app and handler for Caddy Server (https://caddyserver.com/)
+// allowing it to turn any handler into one supporting the Vulcain protocol.
+
+package caddy
+
+import (
+	"errors"
+	"log/slog"
+
+	outline_prometheus "github.com/Jigsaw-Code/outline-ss-server/prometheus"
+	outline "github.com/Jigsaw-Code/outline-ss-server/service"
+	"github.com/caddyserver/caddy/v2"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
```

Comment:
This kind of global registration is bad practice. This is really application code that should be called from main instead.

No need to change for this PR, since that seems to be what people do for Caddy, but consider separating the module implementation and the integration into Caddy's main.